File: feature-favorite-venues/src/main/java/home/oleg/feature_favorite_venues/di/FavoriteVenuesFragmentModule.java
package home.oleg.feature_favorite_venues.di;

import android.arch.lifecycle.ViewModelProvider;
import android.arch.lifecycle.ViewModelProviders;

import com.smedialink.feature_add_favorite.CreateFavoriteViewModel;

import dagger.Module;
import dagger.Provides;
import home.oleg.feature_favorite_venues.FavoritePlacesFragment;
import home.oleg.feature_favorite_venues.FavoritePlacesViewModel;
import home.oleg.feature_favorite_venues.FavoriteVenuesAdapter;
import home.oleg.placesnearme.core_presentation.recyclerview.VenuesDiffCallback;
import io.reactivex.annotations.NonNull;

@Module
public abstract class FavoriteVenuesFragmentModule {

    @Provides
    @NonNull
    static FavoriteVenuesAdapter provideFavoriteVenuesAdapter(
            FavoritePlacesFragment fragment) {
        return new FavoriteVenuesAdapter(VenuesDiffCallback.VENUES_DIFF_CALLBACK, fragment);
    }

    @Provides
    @NonNull
    static FavoritePlacesViewModel provideFavoritePlacesViewModel(
            FavoritePlacesFragment fragment,
            ViewModelProvider.Factory factory) {
        return ViewModelProviders.of(fragment, factory).get(FavoritePlacesViewModel.class);
    }

    @Provides
    @NonNull
    static CreateFavoriteViewModel provideCreateFavoriteViewModel(
            FavoritePlacesFragment fragment,
            ViewModelProvider.Factory factory) {
        return ViewModelProviders.of(fragment, factory).get(CreateFavoriteViewModel.class);
    }
}

File: feature-favorite-venues/src/main/java/home/oleg/feature_favorite_venues/di/FavoriteVenuesFragmentModule.kt
package home.oleg.feature_favorite_venues.di

import com.smedialink.feature_add_favorite.presentation.CreateFavoriteViewModelDelegate

import androidx.lifecycle.ViewModelProvider
import dagger.Module
import dagger.Provides
import home.oleg.feature_favorite_venues.presentation.ui.FavoritePlacesFragment
import home.oleg.feature_favorite_venues.presentation.FavoritePlacesViewModel
import home.oleg.feature_favorite_venues.presentation.ui.FavoriteVenuesAdapter
import home.oleg.placesnearme.core_presentation.extensions.viewModel
import home.oleg.placesnearme.core_presentation.recyclerview.ItemsDiffCallback

@Module
object FavoriteVenuesFragmentModule {

    @JvmStatic
    @Provides
    internal fun provideFavoriteVenuesAdapter(
            fragment: FavoritePlacesFragment): FavoriteVenuesAdapter {
        return FavoriteVenuesAdapter(ItemsDiffCallback.ITEMS_DIFF_CALLBACK, fragment)
    }

    @JvmStatic
    @Provides
    internal fun provideFavoritePlacesViewModel(
            fragment: FavoritePlacesFragment,
            factory: ViewModelProvider.Factory): FavoritePlacesViewModel = fragment.viewModel(factory)

}
